File: 2470-removing-stars-from-a-string/solution.java
class Solution {
    public String removeStars(String s) {
        StringBuilder sb = new StringBuilder();
        
        for (char c : s.toCharArray()) {
            if (c == '*') {
                // פעולת ה-Pop: מוחקים את התו האחרון שהוספנו
                if (sb.length() > 0) {
                    sb.deleteCharAt(sb.length() - 1);
                }
            } else {
                // פעולת ה-Push: מוסיפים תו למחסנית
                sb.append(c);
            }
        }
        
        return sb.toString();
    }
}

File: 2470-removing-stars-from-a-string/solution.py
class Solution:
    def removeStars(self, s: str) -> str:
        stack = []
        counter=0
        for char in s:
            stack.append(char)
        str = ""
        while stack:
            c = stack.pop()
            if c != '*' and counter==0:
                str=c+str
            elif c!= '*' and counter>0:
                counter-=1
            elif c=='*':
                counter+=1                
        return str

